src/swap/saga.ts

```diff
@@ -299,74 +268,71 @@ export function* swapSubmitSaga(action: PayloadAction<SwapInfo>) {
+          return {
             context: swapApproveContext,
             __typename: 'TokenApproval',
             networkId,
             type: TokenTransactionTypeV2.Approval,
-            transactionHash: approvalTxHash,
+            transactionHash: txHash,
             tokenId: fromToken.tokenId,
             approvedAmount,
             feeCurrencyId,
-          })
-        )
+          }
+        }
+        createSwapStandbyTxHandlers.push(createApprovalStandbyTx)
       }
     }
 
-    const swapTxHash = txHashes[txHashes.length - 1]
-    yield* put(
-      addStandbyTransaction({
-        context: swapExecuteContext,
-        __typename: 'TokenExchangeV3',
-        networkId,
-        type: TokenTransactionTypeV2.SwapTransaction,
-        inAmount: {
-          value: swapAmount[Field.TO],
-          tokenId: toToken.tokenId,
-        },
-        outAmount: {
-          value: swapAmount[Field.FROM],
-          tokenId: fromToken.tokenId,
-        },
-        transactionHash: swapTxHash,
-        feeCurrencyId,
-      })
+    const createSwapStandbyTx = (
+      txHash: string,
+      feeCurrencyId?: string
+    ): BaseStandbyTransaction => ({
+      context: swapExecuteContext,
+      __typename: 'TokenExchangeV3',
+      networkId,
+      type: TokenTransactionTypeV2.SwapTransaction,
+      inAmount: {
+        value: swapAmount[Field.TO],
+        tokenId: toToken.tokenId,
+      },
+      outAmount: {
+        value: swapAmount[Field.FROM],
+        tokenId: fromToken.tokenId,
+      },
+      transactionHash: txHash,
+      feeCurrencyId,
+    })
+    createSwapStandbyTxHandlers.push(createSwapStandbyTx)
+
+    const txHashes = yield* call(
+      sendTransactionsSaga,
+      serializablePreparedTransactions,
+      networkId,
+      createSwapStandbyTxHandlers
     )
+
+    Logger.debug(TAG, 'Successfully sent swap transaction(s) to the network', txHashes)
+
     navigate(Screens.WalletHome)
     submitted = true
 
-    const swapTxReceipt = yield* call([publicClient[network], 'waitForTransactionReceipt'], {
-      hash: swapTxHash,
-    })
-    Logger.debug('Got swap transaction receipt', swapTxReceipt)
-    trackedTxs[trackedTxs.length - 1].txReceipt = swapTxReceipt
-
-    // Also get the receipt of the first transaction (approve), for tracking purposes
-    try {
-      if (txHashes.length > 1) {
-        const approveTxReceipt = yield* call([publicClient[network], 'getTransactionReceipt'], {
-          hash: txHashes[0],
-        })
-        Logger.debug('Got approve transaction receipt', approveTxReceipt)
-        trackedTxs[0].txReceipt = approveTxReceipt
-      }
-    } catch (e) {
-      Logger.warn(TAG, 'Error getting approve transaction receipt', e)
+    // wait for the tx receipts, so that we can track them
+    for (let i = 0; i < txHashes.length; i++) {
+      const txReceipt = yield* call([publicClient[network], 'waitForTransactionReceipt'], {
+        hash: txHashes[i],
+      })
```

Comment:
Should we do this in parallel to avoid unnecessary wait?
Well now that I'm writing this, this probably doesn't matter since there's no user facing wait here.

Reply:
yeah i'd prefer to leave it like this because it's a background task and easier to read this way